controllers/appgroup_controller.go

```diff
@@ -71,11 +71,12 @@ func (r *ApplicationGroupReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	}
 	patch := client.MergeFrom(appGroup.DeepCopy())
 
-	statusHelper := helpers.StatusHelper{
-		Client:    r.Client,
-		Logger:    logr,
-		PatchFrom: patch,
-		Recorder:  r.Recorder,
+	statusHelper := &helpers.StatusHelper{
```

Comment:
The helpers get instantiated every single time on reconciliation of each ApplicationGroup instance. The only variable being the `patch` object.
Might make sense to instantiate this once and make it part of the `ApplicationGroupReconciler` instance?
Instead of the helper methods using the member `patch` object, they can accept the `patch` as an input.

Also see comment below for `ReconcileHelper`

Reply:
See comment below for my opinion on this